hlsl-headers/include/vk/khr/cooperative_matrix.hlsli

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef VULKAN_HLSL_SPV_KHR_COOPERATIVE_MATRIX_H_
+#define VULKAN_HLSL_SPV_KHR_COOPERATIVE_MATRIX_H_
+
+
+// TODO: Add a macro to HLSL to be able to check the Vulkan version being
+// targeted.
+
+#define SPV_KHR_CooperativeMatrix                  \
+  vk::ext_extension("SPV_KHR_cooperative_matrix"), \
+      vk::ext_capability(/* CooperativeMatrixKHRCapability */ 6022)
+
+namespace vk {
+namespace khr {
+
+template <typename ComponentType, uint scope, uint rows, uint columns, uint use>
```

Comment:
We can come back to this later. I have used "enum" instead of "enum class", so I can use the name of the enum in the template. We can discuss if we should use "enum class" later. Still fix the issue though.